File: libro/133_cadenas.cpp
/* ********************************************************
* Programa: Cadenas
*
* Descripción:
*  Ejemplo de manejo de cadenas de caracteres
******************************************************** */

#include <stdio.h>
#include <strings.h>

typedef char TipoCadena[20];

int main () {
  TipoCadena nombre, apellido;
  typedef char TipoTexto[100];
  TipoTexto texto;
  int resul;

  printf( "Nombre y Apellido? " );
  scanf( "%s%s", nombre, apellido );
  printf( "Datos: %s - %s \n", nombre, apellido );
  printf( "Longitudes: %4d%4d\n", strlen(nombre), strlen(apellido) );
  strcpy( texto , nombre ) ;
  printf( "Texto copiado: %s\n", texto ) ;
  strcat( texto, apellido );
  printf( "Texto concatenado: %s\n", texto) ;
  resul = strcmp( apellido, nombre );
  if (resul == 0) {
    printf( "Nombre y Apellido iguales\n" );
  } else if (resul > 0) {
    printf( "%s es anterior a %s\n", nombre, apellido );
  }
}

File: libro/133_cadenas.py
# Programa: Cadenas
# Descripción:
#   Ejemplo de manejo de cadenas de caracteres


# Programa principal
if __name__ == "__main__":
  nombre = ""
  apellido = ""
  texto = ""
  resul = 0

  (nombre, apellido) = input("Nombre y Apellido?").split()
  print("Datos: {} - {}".format(nombre, apellido))
  print("Longitudes: {0:4d} {1:4d}".format(len(nombre), len(apellido)))
  texto = nombre 
  print("Texto copiado: {}".format(texto))
  texto = texto + apellido
  print("Texto concatenado: {}".format(texto))

  if (apellido == nombre):
    print("Nombre y Apellido iguales")
  else:
    palabras = [nombre, apellido]
    palabras.sort()
    print("{} es anterior a {}".format(palabras[0], palabras[1]))

